packages/connectivity/src/scp-cf/destination/destination-service.ts

```diff
@@ -150,26 +124,47 @@ export interface AuthAndExchangeTokens {
 }
 
 /**
- * Fetches a specific destination by name from the given URI, including authorization tokens.
- * For destinations with authenticationType OAuth2SAMLBearerAssertion, this call will trigger the OAuth2SAMLBearerFlow against the target destination.
- * In this pass the access token as string.
- * Fetches a specific destination with authenticationType OAuth2UserTokenExchange by name from the given URI, including authorization tokens.
- * @param destinationServiceUri - The URI of the destination service
- * @param token - The access token or AuthAndExchangeTokens if you want to include the X-user-token for OAuth2UserTokenExchange.
- * @param options - Options to use by retrieving destinations
- * @returns A Promise resolving to the destination
  * @internal
  */
 export async function fetchDestination(
+  destinationName: string,
   destinationServiceUri: string,
-  token: string | AuthAndExchangeTokens,
-  options: DestinationServiceOptions
-): Promise<Destination> {
-  return fetchDestinationByTokens(
-    destinationServiceUri,
-    typeof token === 'string' ? { authHeaderJwt: token } : token,
-    options
-  );
+  jwt: string
+): Promise<DestinationsByType> {
+  const targetUri = `${removeTrailingSlashes(
+    destinationServiceUri
+  )}/destination-configuration/v1/destinations/${destinationName}?$skipTokenRetrieval=true`;
+
+  const headers = wrapJwtInHeader(jwt).headers;
+  try {
+    const response = await callDestinationEndpoint(
+      { uri: targetUri, tenantId: getTenantFromTokens(jwt) },
+      headers
+    );
+    const destination = parseDestination(
+      response.data.destinationConfiguration
+    );
+
+    return {
+      instance: response.data.owner?.InstanceId ? [destination] : [],
+      subaccount: response.data.owner?.SubaccountId ? [destination] : []
+    };
+  } catch (err) {
+    if (
+      err.response?.status === 404 &&
+      err.response?.data?.ErrorMessage ===
+        'Configuration with the specified name was not found'
+    ) {
+      return {
+        instance: [],
+        subaccount: []
+      };
+    }
+    throw new ErrorWithCause(
+      `Failed to fetch destination.${errorMessageFromResponse(err)}`,
+      err
+    );
+  }
```

Comment:
Is my understanding correct that you've changed the implementation to "standalone" instead of calling `fetchDestinationByTokens`? The difference that stands out to me is `?$skipTokenRetrieval=true`. Can you provide some context what the idea here is?

Also I don't understand what is meant by "fetching destination _by token_". Could you explain this to me please?